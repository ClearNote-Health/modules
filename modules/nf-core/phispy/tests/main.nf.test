// nf-core modules test phispy
nextflow_process {

    name "Test Process PHISPY"
    script "../main.nf"
    process "PHISPY"

    tag "modules"
    tag "modules_nfcore"
    tag "phispy"

    test("bacteroides_fragilis - gbff") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file( params.modules_testdata_base_path + "/genomics/prokaryotes/bacteroides_fragilis/genome/genome.gbff.gz", checkIfExists: true )
                ]
                """
            }
        }

        then {
            def log_text = path(process.out.log[0][1]).text
            def tsv_text = path(process.out.coordinates[0][1]).text
            assertAll(
                { assert process.success },
                { assert log_text.contains('NZ_CP069563.1') },
                { assert log_text.contains('91') },
                { assert log_text.contains('Kept') },
                { assert tsv_text.contains('NZ_CP069563.1') },
                { assert tsv_text.contains('pp1') },
                { assert tsv_text.contains('Longest Repeat flanking phage and within 2000 bp') },
                { assert path(process.out.gbk[0][1]).linesGzip.toString().contains('5150703 bp') },
                { assert snapshot(process.out.versions).match() }
            )
        }

    }

    test("bacteroides_fragilis - gbff - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file( params.modules_testdata_base_path + "/genomics/prokaryotes/bacteroides_fragilis/genome/genome.gbff.gz", checkIfExists: true )
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match() }
            )
        }

    }

}