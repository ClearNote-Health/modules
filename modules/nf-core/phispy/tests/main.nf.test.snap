{
    "bacteroides_fragilis - gbff - stub": {
        "content": [
            [
                "versions.yml:md5,d26f11b7160b164c45b7e1acc4c667b6"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-19T14:35:03.378235023"
    },
    "bacteroides_fragilis - gbff": {
        "content": [
            [
                "versions.yml:md5,d26f11b7160b164c45b7e1acc4c667b6"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-19T14:34:56.397801796"
    }
}